src/routerd_lib/handlers/proxy.cpp

```diff
@@ -94,40 +151,98 @@ namespace NAC {
                         ProcessServiceResponse(request, response, service.Name, response.get());
                     }
 
-                    Iter(request, args);
+                    Iter(request, args); // recursion depth is limited by graph size, which is small.
                 });
 
-                if (!rv) {
+                if (!rv) { // could not connect
                     failedServices.push_back(service.Name);
                     continue;
                 }
 
-                auto msg = request->OutgoingRequest(service.Path, args);
-                msg.Memorize(request);
-
                 request->NewRequest(service.Name);
-                rv->PushWriteQueueData(msg);
+
+#ifdef AC_DEBUG_ROUTERD_PROXY
+                print_outgoing_request(request);
+#endif
+                // schedule payload to be sent to connected service
+                if (!service.SendRawOutputOf.empty()) {
+                    auto &&outgoing_request = request->GetOutGoingRequest();
+                    auto matching_part = outgoing_request.PartByName(service.SendRawOutputOf);
+                    if (matching_part) {
+#ifdef AC_DEBUG_ROUTERD_PROXY
+                        std::cerr << "to service " << service.Name
+                                  << " will send_raw_output_of " << service.SendRawOutputOf << std::endl;
+#endif
+                        rv->PushWriteQueueData(matching_part->GetBody());
+                    }
+                    else { // should not happen: we've checked it above
```

Comment:
"}" and "else" should be on the same line, separated by empty line from previous expression.